lib/iris/coords.py

```diff
@@ -2114,22 +2114,38 @@ def collapsed(self, dims_to_collapse=None):
         if np.issubdtype(self.dtype, np.str_):
             # Collapse the coordinate by serializing the points and
             # bounds as strings.
-            def serialize(x):
-                return "|".join([str(i) for i in x.flatten()])
+            def serialize(x, axis):
+                if axis is None:
+                    return "|".join(str(i) for i in x.flatten())
```

Comment:
I think it's worth discussing if it's wise to allow collapsing over multiple dimensions (as supported by `x.flatten()` here). I believe the original case in #3653 only collapses one dimension. One thing worth considering is that with the old behaviour it ough to be possible to reconstruct the original coordinate from the collapsed coordinate, but this is broken when collapsing multiple dimensions. I suppose this boils down to if we interpret `"|"` to imply a positional relation or if we consider it to be "commutative".

If we don't want to handle collapsing multiple dimensions then an appropriate error message would be good here. If we're happy handling collapsing multiple dimensions then I think it would be worth adding a test which demonstrated this behaviour.

Reply:
I think it's preferable to support it even if the output coord is less informative than the single dimension case:  if we raise here then a user has to remove the coordinate before they can perform their collapsing operation, so they will have even less information on the output cube.  They may well not even care about this particular coordinate but just see "I have to remove this coord" as an annoying extra step to get where they want to go.

Happy to add an extra test.

Reply:
Actually I think this is already covered by [test_serialize](https://github.com/SciTools/iris/blob/1bfe3113cb6c1460886ce07dfe495c32782f1aa8/lib/iris/tests/unit/coords/test_Coord.py#L437-L476)?

Reply:
It looks like the 2D string arrays in that test are for the bounds (for which only the slices are serialized), the points array being serialized is still 1D. I think there's still room for a more explicit test here.

Reply:
https://github.com/SciTools/iris/pull/5955/commits/638d4910ae05b4efa32c36ed4f58ffa2491a9517